assets/app/scripts/directives/logViewer.js

```diff
@@ -27,7 +27,15 @@ angular.module('openshiftConsole')
         // this webkit bug with user-select: none;
         //   https://bugs.webkit.org/show_bug.cgi?id=80159
         line.firstChild.setAttribute('data-line-number', lineNumber);
-        line.lastChild.appendChild(document.createTextNode(text));
+
+        // Escape ANSI color codes
+        var escaped = ansi_up.escape_for_html(text);
+        var html = ansi_up.ansi_to_html(escaped);
+        var linkifiedHTML = ansi_up.linkify(html);
+        line.lastChild.innerHTML = linkifiedHTML;
```

Comment:
sounds good to me then

On Tue, Mar 22, 2016 at 4:28 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/directives/logViewer.js
> <https://github.com/openshift/origin/pull/7922#discussion_r57062890>:
>
> > @@ -27,7 +27,15 @@ angular.module('openshiftConsole')
> >          // this webkit bug with user-select: none;
> >          //   https://bugs.webkit.org/show_bug.cgi?id=80159
> >          line.firstChild.setAttribute('data-line-number', lineNumber);
> > -        line.lastChild.appendChild(document.createTextNode(text));
> > +
> > +        // Escape ANSI color codes
> > +        var escaped = ansi_up.escape_for_html(text);
> > +        var html = ansi_up.ansi_to_html(escaped);
> > +        var linkifiedHTML = ansi_up.linkify(html);
> > +        line.lastChild.innerHTML = linkifiedHTML;
>
> I'm not seeing any performance problems whatsoever. Autoscrolling takes
> more than 20x as long as adding the log content even with ansi_up.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/7922/files/f124b83ddcf1f48edf614897fc6c34eb17a333aa#r57062890>
>